### DISCUSSION TITLE
Is it the proper way of mounting a livewire component when the modal is open ?

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#7681] [4. General]
Hi, 

I have this simplified modal view : 
```blade
<div x-data="{ showModal: false }">
    <!-- Button to open the modal -->
    <button @click="showModal = true">Open Modal</button>

    <!-- Modal -->
    <div x-show="showModal" @click.away="showModal = false">
        <!-- Dynamically load Livewire component -->
        <template x-if="showModal">
            @livewire('edit-image-metadata', ['mediaId' => $mediaId])
        </template>
    </div>
</div>
```

Is it the proper way of mounting/unmounting the Livewire component edit-image-metadata ? In the DOM, the #document-fragment stays, but the HTML nodes below disappear. 

Because the content is already in the DOM inside the fragment, the content shows immediately. Is there a way to fetch the content only when the modal opens ? Because it seems that the content was already fetched at the page load and put in a fragment. 

Thank you 🙏

**Comment by joshhanley** [comment]
@joffreypersia that probably won't work as Livewire component tags are evaluated by Blade on the server, where as using `x-if` and a template tag is only front end.

I would just use a blade conditional `@if` to display the component and have some form of loading state that gets shown in the model while you make a request to the server to load the contents of the modal.

**Reply by joffreypersia** [reply]
yeah thank you Josh, but I am a bit lost to write it. If I make a php `@if`, how can I communicate with the PHP variable to change his state when the user will click on the button ? 
Should I have to use a LV modal ? Because mine is written in Blade (same as the screencast from Caleb)

Thank you 🙏

**Comment by mcpuishor** [comment]
If the issue is performance related when loading the modal inside the component, have you tried making the component `lazy` and have a placeholder to be rendered on the initial load? 

In theory, the actual `render` method of the component 'hosted' inside the modal should be called when the component gets into view.

**Reply by joffreypersia** [reply]
Oh perfect ! It worked. Livewire is magical. 
Thank you !
